cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -1005,6 +1043,27 @@ impl ClientTrait for MockClient {
             _ => panic!("expected get_store_info response, found {:?}", &mock_resp),
         }
     }
+
+    async fn is_publish_complete(
+        &self,
+        _store_paths: &[String],
+    ) -> Result<bool, CatalogClientError> {
+        let mock_resp = self
+            .mock_responses
+            .lock()
+            .expect("couldn't acquire mock lock")
+            .pop_front();
+        let statuses = match mock_resp {
+            Some(Response::GetStorepathStatus(resp)) => resp,
+            _ => panic!("expected get_store_info response, found {:?}", &mock_resp),
+        };
+        let all_narinfo_available = statuses.items.values().all(|storepath_statuses_for_drv| {
+            storepath_statuses_for_drv
+                .iter()
+                .all(|status| status.narinfo_known)
+        });
+        Ok(all_narinfo_available)
+    }
```

Comment:
nit: we could probably just panic unless we're actually using this for any tests